### DISCUSSION TITLE
[@types/pg] The last overload gave the following error.

### DISCUSSION THREAD

**Discussion by will-path** [DefinitelyTyped/DefinitelyTyped#69262] [Issues with a @types package] [answered]
Starting with v8.11.3, calling `query` like this:
```typescript
await client.query(text, values);
```
Where `values` type is an array of a union type, will sometimes fail a type-check, with this error message:

```
  The last overload gave the following error.
    Argument of type '(string | Date | null)[]' is not assignable to parameter of type '(err: Error, result: QueryResult<any>) => void'.
      Type '(string | Date | null)[]' provides no match for the signature '(err: Error, result: QueryResult<any>): void'.
```
It _should_ match the function overload here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pg/index.d.ts#L231
But instead it looks like it's matching the second to last one.

[a comment by an automated account is not shown]

**Comment by ghost** [comment, marked as answer]
Will be fixed by PR #69053
